Documentation/scalability/scalability.rst

```diff
@@ -0,0 +1,397 @@
+
+.. figure:: images/image_5_02.png
+
+As we have increased the number of nodes in the cluster to 1000 it is expected
+to see a small growth of the memory usage in all metrics. However, it is
+relevant to point out that **the number of nodes does not have a significant
+growth in Cilium’s memory consumption in both control and dataplane.**
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+6. Deploy 25 more deployments on each namespace
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This will now bring us a total of
+``5 namespaces * (25 old deployments + 25 new deployments)=250`` deployments in
+the entire cluster.
+We did not install 250 deployments from the start since we only had 2 nodes and
+that would create 125 pods on each worker node and accordingly with the
+Kubernetes documentation the maximum recommended number of pods per node is 100.
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+7. Scale each deployment to 200 replicas (50000 pods in total)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Having 5 namespaces with 50 deployments means that we have 250 different unique
+security identities. Having a low cardinality in the labels selected by Cilium
+helps the scalability of security identities in the cluster. By default, Cilium
+has a limit of 16k security identities, however this limit can be incremented by
+the user with ``bpf-policy-map-max`` in the Cilium ``ConfigMap``.
+
+.. figure:: images/image_7_01.png
+
+The first mark represents the action of scaling up the deployments, the second
+mark represents when 50000 pods were in ready state.
+
+* It is expected to see the CPU usage of Cilium increasing since each Cilium
+  agent receives events from Kubernetes when a new pod is scheduled and started
+  in the node where that Cilium pod is running.
+
+* The average CPU consumption of all Cilium agents was 3.38% on a 2vCPU machine.
+  At one given point, roughly around minute 15:23, one of those Cilium agents
+  picked 27.94% CPU usage.
+
+* Cilium Operator had a stable 5% CPU consumption while the pods were being
+  created.
+
+.. figure:: images/image_7_02.png
+
+Similar to the behavior seen with the growth number of nodes, increasing pods
+also increases Cilium memory consumption.
+
+* As we increased the number of pods from 250 to 50000 we saw a max memory usage
+  of 573MiB for one of the Cilium agents while the average was 438 MiB.
+* For the BPF memory usage we saw a max usage of 462.7MiB
+* This means that each **Cilium agent memory increases 10.5KiB per pod created
+  in the cluster.**
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+8. Deploy 250 policies for 1 namespace
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Here we have created 125 L4 network policies and 125 L7 policies. Each policy
+selected all pods on this namespace and was allowed to send traffic to another
+pod on this namespace. Each 250 policies differ on the chosen port. In the end
```

Comment:
maybe something like:

Each of the 250 policies allows access to a disjoint set of ports.